activerecord/test/cases/validations/uniqueness_validation_test.rb

```diff
@@ -427,4 +428,18 @@ def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
     assert reply.valid?
     assert topic.valid?
   end
+
+  def test_validate_uniqueness_of_custom_primary_key
+    Keyboard.validates_uniqueness_of(:key_number)
```

Comment:
This test will leak state, as you've modified the class. Please create an anonymous class for use in the test.